crates/sargon/src/wrapped_radix_engine_toolkit/low_level/manifest_summary/account_deposit/simple_non_fungible_resource_bounds.rs

```diff
@@ -0,0 +1,195 @@
+use crate::prelude::*;
+
+/// Represents the bounds for a simple non-fungible resource, which can be either exact or not exact.
+#[derive(Clone, Debug, PartialEq, Eq, uniffi::Enum)]
+pub enum SimpleNonFungibleResourceBounds {
+    Exact {
+        amount: Decimal,
+        certain_ids: Vec<NonFungibleLocalId>,
+    },
+    NotExact {
+        certain_ids: Vec<NonFungibleLocalId>,
+        lower_bound: LowerBound,
+        upper_bound: UpperBound,
+        allowed_ids: AllowedIds,
+    },
+}
+
+impl SimpleNonFungibleResourceBounds {
+    pub fn exact(
+        amount: Decimal,
+        certain_ids: impl IntoIterator<Item = NonFungibleLocalId>,
+    ) -> Self {
+        Self::Exact {
+            amount,
+            certain_ids: certain_ids.into_iter().collect(),
+        }
+    }
+
+    pub fn not_exact(
+        certain_ids: impl IntoIterator<Item = NonFungibleLocalId>,
+        lower_bound: LowerBound,
+        upper_bound: UpperBound,
+        allowed_ids: AllowedIds,
+    ) -> Self {
+        Self::NotExact {
```

Comment:
You can `implement Deref<Decimal> for Lowerbound` and same for `Upperbound` then you can add an assert here, we wanna assert that `lower_bound < upper_bound` right, so you would then use `Deref` like so:
`assert!(*lower_bound <= *upper_upperbound, "Expected lower to <= upper"); and document panic and add unit test :)